econml/inference/_inference.py

```diff
@@ -690,43 +688,44 @@ def summary_frame(self, alpha=0.1, value=0, decimals=3,
         treatment_names = self.treatment_names if treatment_names is None else treatment_names
         output_names = self.output_names if output_names is None else output_names
         to_include = OrderedDict()
-        to_include['point_estimate'] = self._array_to_frame(self.d_t, self.d_y, self.point_estimate,
-                                                            output_names=output_names, treatment_names=treatment_names)
+        to_include['point_estimate'] = self._reshape_array(self.point_estimate)
+        nx = to_include['point_estimate'].shape[0] // self.d_t // self.d_y
```

Comment:
Is there not a more direct way of getting this than dividing by T and Y?  Seems a bit weird, but ok if there's no clean alternative.

Reply:
yeah, since the shape of `const_marginal_effect` is (m, dy, dt) but the shape of coef is (dy, dt, m) (m here means # of x features), I don't find an easier way to extract m when they are in different shape position

Reply:
To me, it seems like

```suggestion
        nx = self.point_estimate.shape[0 if self.inf_type == 'coefficient' else -1]
```

would be clearer than doing arithmetic to pull it out.

Reply:
(perhaps with a comment about why it's necessary)

Reply:
I am slightly worried about potential collapsed dimension issue in your suggested changes, and also it's hard coded. I am inclined to keep my current implementation and add comment around this line.